sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/azurite/AzuriteFixture.java

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.test.shared.azurite;
+
+import com.azure.core.util.Configuration;
+import com.azure.storage.common.test.shared.TestAccount;
+
+import java.io.BufferedReader;
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.Comparator;
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class manages Azurite Lifecycle for a test class.
+ * - Creates accounts pool, so that each test has own account, bump up pool size if you're running out of accounts
+ * - Starts Azurite process
+ *  - Tears down Azurite process after test class is run
+ *  It requires Azurite V3. See instalation insturctions here https://github.com/Azure/Azurite.
+ *  After installing Azuirte define env variable AZURE_AZURITE_LOCATION that points to azurite installation (e.g. C:\Users\kasobol.REDMOND\AppData\Roaming\npm)
+ *  NodeJS installation is also required and node should be in the $PATH.
+ */
+public class AzuriteFixture implements Closeable {
+
+    private static final String AZURITE_LOCATION_KEY = "AZURE_AZURITE_LOCATION";
+
+    private final Path tempDirectory;
+    private final Process process;
+    private volatile int blobsPort;
+    private volatile int queuesPort;
+    private final CountDownLatch portReaderGate = new CountDownLatch(2);
+    private final String accountName = UUID.randomUUID().toString();
+    private final String accountKey = Base64.getEncoder()
+        .encodeToString(UUID.randomUUID().toString().getBytes(StandardCharsets.UTF_8));
+
+    public AzuriteFixture() {
```

Comment:
I'm not sure how I feel with having the test code running a system process and managing the closing of it. Was it looked into requiring Azurite being started externally for the tests to run instead? Be it a user running Azurite locally or the step installing Azurite in CI triggering it to run.

Reply:
I'll give it a try. In dotnet we went with that approach because we wanted more granular isolation between tests.

Reply:
it did work out well.